sdks/python/apache_beam/runners/common.py

```diff
@@ -926,31 +922,16 @@ def _invoke_process_per_window(self,
                                  additional_kwargs,
                                 ):
     # type: (...) -> Optional[SplitResultResidual]
-
     if self.has_windowed_inputs:
       assert len(windowed_value.windows) <= 1
       window, = windowed_value.windows
-      side_inputs = [si[window] for si in self.side_inputs]
-      side_inputs.extend(additional_args)
-      args_for_process, kwargs_for_process = util.insert_values_in_args(
-          self.args_for_process, self.kwargs_for_process,
-          side_inputs)
-    elif self.cache_globally_windowed_args:
-      # Attempt to cache additional args if all inputs are globally
-      # windowed inputs when processing the first element.
-      self.cache_globally_windowed_args = False
-
-      # Fill in sideInputs if they are globally windowed
```

Comment:
Is my understanding correct that the fundamental flaw in this logic is that even for a global window, the sideinput can refresh if a window has a trigger configured?